Constify inKino

jonahwilliams opened the pull request:
I've added `const` where it could be trivially swapped for new.  I did indulge myself a little and swapped around widget construction in `storyline_widget.dart` to make more of the text widgets const.

roughike commented on lib/ui/event_details/storyline_widget.dart:
```diff
@@ -27,29 +27,34 @@ class _StorylineWidgetState extends State<StorylineWidget> {
-          ),
-        ),
-      ));
+      const captionStyle = const TextStyle(
```
Now that this got a little longer, I would prefer to extract this into a new method so that it's easier to follow.

Something like:

```dart
  Widget _buildCaption() {
    ...
    if (_isExpandable) {
      content.add(_buildExpandCollapsePrompt());
    }
    ...
  }

  Widget _buildExpandCollapsePrompt() {
    const captionStyle = const TextStyle(
      fontSize: 12.0,
      fontWeight: FontWeight.w600,
      color: Colors.black54,
    );

    if (_isExpanded) {
      return const Padding(
        padding: const EdgeInsets.only(left: 4.0),
        child: const Text('[touch to collapse]', style: captionStyle),
      );
    }

    return const Padding(
      padding: const EdgeInsets.only(left: 4.0),
      child: const Text('[touch to expand]', style: captionStyle),
    );
  }
```

roughike commented:
Thanks! I nitpicked about one tiny thing.

roughike commented:
Also, is there an analysis option or similar which automatically suggests using const instead of new where appropriate?

jonahwilliams commented:
Yes, I've added an example `analysis_options.yaml` which includes the `prefer_const_constructors` lint.  This will catch all places where you could insert a `const`, turns out I missed a few too.

I've also added one more option in the language setting, this is a config necessary to make flutter work with the linter.  Finally there is a commented out line too: `implicit-dynamic: false` will help make your code dart 2 safe, since it will try and prevent you from accidentally using `dynamic` and passing that to something that expects a more specific type.  This was generally okay in dart 1, but it's not really safe.  A quick example:

```
void takesAMap(Map<String, String> foo) { ... }
...
Map foo = {};
takesAMap(foo); // Okay in dart 1, will fail at runtime in dart 2
```

roughike commented:
Thanks! 

The last commit actually made the Travis build fail, and it fails locally for me as well. Do you have time to either revert it or investigate how to test errors in Futures? If not, I'm happy to continue.

jonahwilliams commented:
That is very odd, I needed that change for the test to pass locally.  Possibly due to slightly different flutter versions?
